rbac/api/cross_access/view.py

```diff
@@ -215,17 +256,30 @@ def validate_and_format_input(self, request_data):
             if not request_data.__contains__(field):
                 self.throw_validation_error("cross-account-request", f"Field {field} must be specified.")
 
-        target_account = request_data.get("target_account")
-        if target_account == self.request.user.account:
-            self.throw_validation_error(
-                "cross-account-request", "Creating a cross access request for your own account is not allowed."
-            )
+        if settings.AUTHENTICATE_WITH_ORG_ID:
+            target_org = request_data.get("target_org")
+            if target_org == self.request.user.org_id:
+                self.throw_validation_error(
+                    "cross-account-request", "Creating a cross access request for your own account is not allowed."
```

Comment:
We should update this to reflect org.
```suggestion
                    "cross-account-request", "Creating a cross access request for your own org is not allowed."
```